examples/05_scaling_non_linear_models.py

```diff
@@ -0,0 +1,537 @@
+# square matrix. To understand where this matrix comes from, we need to delve a
+# little bit deeper these methods internals.
+#
+# .. topic:: Kernel methods
+#
+#    Kernel methods address non-linear problems by the similarities between
+#    each pair of inputs. Using a similarity matrix to solve a machine learning
+#    problem allows to catch complex, non-linear relationships within the data.
+#    But it requires inverting this matrix, which can be a computational burden
+#    when the sample sizes increases.
+
+
+###############################################################################
+# Kernel approximation methods
+# ----------------------------
+# From what was said below, two criterions are limiting a kernel algorithm to
+# scale:
+#
+# * It processes a matrix, whose size increases quadratically with the number
+#   of samples.
+# * During fitting time, this matrix is **inverted**, meaning it has to be
+#   loaded into main memory.
+#
+# Kernel approximation methods such as |RBF| or |NYS| [#nys_ref]_ try to
+# approximate this similarity matrix, without actually creating it. By allowing
+# the program to not compute the perfect similarity matrix, the problem
+# complexity becomes linear! Plus, the samples also do not need to be processed
+# at once into main memory. We are not bound to use a |SVC| anymore, and can
+# instead use an online optimization that will process the input by batch.
+#
+# .. topic:: Online algorithms
+#
+#    An online algorithm [#online_ref]_ is an algorithm that treats its input
+#    piece by piece in a serial fashion. An famous example is the stochastic
+#    gradient descent [#sgd_ref]_, where an estimation the objective function's
+#    gradient is computed on a batch of the data at each step.
+#
+#
+
+###############################################################################
+# Reducing the transformers dimensionality
+# ----------------------------------------
+# There is one last scalability issue in our pipeline: the |SE| and the |RBF|
+# both implement the |fit| method. How to adapt those method to an online
+# setting, where the data is never loaded as a whole?
+#
+# A simple solution is to partially fit the |SE| and the |RBF| on a subset of
+# the data, prior to the online fitting step.
+
+from sklearn.kernel_approximation import RBFSampler
+n_out_encoder = 1000
+n_out_rbf = 5000
+n_samples_encoder = 10000
+
+X_encoder, _ = get_X_y(nrows=n_samples_encoder, names=columns_names)
+
+similarity_encoder = SimilarityEncoder(
+    similarity='ngram', categories='most_frequent', n_prototypes=n_out_encoder,
+    random_state=42, ngram_range=(2, 4))
+
+# Fit the rbf_sampler with the similarity matrix.
+column_transformer = make_column_transformer(
+    (similarity_encoder, ['NONPROPRIETARYNAME']),
+    (OneHotEncoder(handle_unknown='ignore'), ['DOSAGEFORMNAME', 'ROUTENAME']),
+    sparse_threshold=1)
+
+transformed_categories = column_transformer.fit_transform(X_encoder)
+
```

Comment:
Maybe say here (in a small comment on one or two lines) that the gamma is an important hyperparameter that needs to be set on the data.